src/yarpWholeBodyStates.cpp

```diff
@@ -1154,23 +1202,52 @@ bool yarpWholeBodyEstimator::computeWorldRootRotoTranslation(double *q_temp)
       }
       return(true);
   }
-  else
-    
+  else    
     return(false);
 }
-
-bool yarpWholeBodyEstimator::setWorldBaseLinkName(std::string linkName)
+bool yarpWholeBodyEstimator::computeBaseVelocity(double* qj,double* dqj)
 {
+  
   if(wholeBodyModel!=NULL)
   {
-    yInfo()<<"Reference link set as world was "<<robot_reference_frame_link;
-    wholeBodyModel->getFrameList().idToIndex(linkName.c_str(),robot_reference_frame_link);
-    yInfo()<<", now it is set to "<<robot_reference_frame_link<<"\n\n\n";
+    int dof = estimates.lastQ.size();
+   // new (&dqjVect) Eigen::Map<Eigen::VectorXd>(dqj,dof);
+    
+    Eigen::Map<Eigen::VectorXd> dqjVect(dqj,dof);
+    Eigen::Matrix<double,6,Eigen::Dynamic,Eigen::RowMajor> complete_jacobian(6,dof+6), joint_jacobian(6,dof),floatingBase_jacobian(6,6);
+    Eigen::Matrix<double,6,Eigen::Dynamic,Eigen::RowMajor> tempMatForComputation(6,dof);//, minusTemp(6,dof);
+    Eigen::VectorXd dvbVect(6);
+    Eigen::Map<Eigen::VectorXd> rotationalVelocityWrapper(estimates.lastBaseVel.data(), estimates.lastBaseVel.size());
```

Comment:
Can we move all this objects to be members of the class instead of local objects? In this way we allocate the memory for them once, instead of continuously allocate memory at each control loop, that is quite time consuming.